core/templates/dev/head/pages/exploration_editor/ExplorationMetadataFormDirective.js

```diff
@@ -0,0 +1,146 @@
+     controller: [
+      '$scope', '$rootScope', 'explorationObjectiveService',
+      'explorationTitleService', 'explorationCategoryService',
+      'explorationStatesService', 'ALL_CATEGORIES',
+      'explorationInitStateNameService', 'explorationLanguageCodeService',
+      'explorationTagsService', 'EXPLORATION_TITLE_INPUT_FOCUS_LABEL',
+      function($scope, $rootScope, explorationObjectiveService,
+       explorationTitleService, explorationCategoryService,
+       explorationStatesService, ALL_CATEGORIES,
+       explorationInitStateNameService, explorationLanguageCodeService,
+       explorationTagsService, EXPLORATION_TITLE_INPUT_FOCUS_LABEL) {
+        $scope.stateNames = explorationStatesService.getStateNames();
+        $scope.TAG_REGEX = GLOBALS.TAG_REGEX;
+        $scope.EXPLORATION_TITLE_INPUT_FOCUS_LABEL =
+         EXPLORATION_TITLE_INPUT_FOCUS_LABEL;
+        $scope.CATEGORY_LIST_FOR_SELECT2 = [];
+        $scope.explorationTitleService = explorationTitleService;
+        $scope.explorationObjectiveService =
+         explorationObjectiveService;
+        $scope.explorationCategoryService =
+         explorationCategoryService;
+        $scope.explorationLanguageCodeService = (
+         explorationLanguageCodeService);
+        $scope.explorationTagsService = explorationTagsService;
+
+        $scope.objectiveHasBeenPreviouslyEdited = (
+         explorationObjectiveService.savedMemento.length > 0);
+
+        $scope.requireTitleToBeSpecified = (
+         !explorationTitleService.savedMemento);
+        $scope.requireObjectiveToBeSpecified = (
+         explorationObjectiveService.savedMemento.length < 15);
+        $scope.requireCategoryToBeSpecified = (
+         !explorationCategoryService.savedMemento);
+        $scope.askForLanguageCheck = (
+         explorationLanguageCodeService.savedMemento ===
+         constants.DEFAULT_LANGUAGE_CODE);
+        $scope.askForTags = (
+         explorationTagsService.savedMemento.length === 0);
+
+
+        for (var i = 0; i < ALL_CATEGORIES.length; i++) {
+          $scope.CATEGORY_LIST_FOR_SELECT2.push({
+            id: ALL_CATEGORIES[i],
+            text: ALL_CATEGORIES[i]
+          });
+        }
+
+        if (explorationStatesService.isInitialized()) {
+          var categoryIsInSelect2 = $scope.CATEGORY_LIST_FOR_SELECT2
+            .some(
+              function(categoryItem) {
+                return categoryItem.id ===
+                 explorationCategoryService.savedMemento;
+              }
+            );
+
+          // If the current category is not in the dropdown, add it
+          // as the first option.
+          if (!categoryIsInSelect2 &&
+           explorationCategoryService.savedMemento) {
+            $scope.CATEGORY_LIST_FOR_SELECT2.unshift({
+              id: explorationCategoryService.savedMemento,
+              text: explorationCategoryService.savedMemento
+            });
+          }
+        }
+        if ($scope.settingsTab) {
```

Comment:
I am concerned about this, but it does highlight a fundamental issue which is that, in the settings tab, edits are saved immediately and in the modal, edits are saved on clicking the "save" button. This suggests that our modelling here might be wrong. Let's arrange a chat to discuss this.